lib/rust/ensogl/example/cached-shape/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+//! Example scene showing the usage of display object auto-layout.
+
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+#![allow(clippy::bool_to_int_with_if)]
+#![allow(clippy::let_and_return)]
+// FIXME: remove
+#![feature(local_key_cell_methods)]
+
+use ensogl_core::display::material::Material;
+use ensogl_core::display::shape::*;
+use ensogl_core::display::world::*;
+use ensogl_core::prelude::*;
+
+use ensogl_core::display::navigation::navigator::Navigator;
+use ensogl_core::display::texture::FloatSampler;
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+mod icon1 {
+    use super::*;
+    ensogl_core::cached_shape! { 32 x 32;
+        () {
+            let shape = Circle(16.px()).fill(color::Rgba::green());
+            shape.into()
+        }
+    }
+}
+
+mod icon2 {
+    use super::*;
+    ensogl_core::cached_shape! { 202 x 312;
+        () {
+            let shape = Rect((200.px(), 310.px())).fill(color::Rgba::red());
+            shape.into()
+        }
+    }
+}
+
+
+
+// ======================
+// === TexturePreview ===
+// ======================
+
+
+#[derive(Debug, Clone, Copy)]
+#[allow(missing_docs)]
+pub struct TextureSystemData {}
+
+impl TextureSystemData {
+    /// Defines a default material of this system.
+    fn material() -> Material {
+        let mut material = Material::new();
+        let shader = "output_color = vec4(input_uv.x, input_uv.y, 0.0, 1.0); output_id=vec4(0.0,0.0,0.0,0.0);";
+        let shader = "output_color = texture(input_pass_cached_shapes,input_uv); output_id=vec4(0.0,0.0,0.0,0.0);";
```

Comment:
I understand that thsi is a debug thing that will be removed in next PR? If so, add comment aobut it pls.